include/seqan3/io/alignment_file/format_sam_base.hpp

```diff
@@ -447,165 +447,175 @@ inline void format_sam_base::read_header(stream_view_type && stream_view,
+                {
+                    static_assert(!detail::is_type_specialisation_of_v<decltype(hdr.ref_ids()), std::deque>,
+                                "The range over reference ids must be of type std::deque such that "
+                                "pointers are not invalidated.");
+
+                    hdr.ref_ids().push_back(id);
+                    hdr.ref_id_info.push_back(info);
+                    hdr.ref_dict[(hdr.ref_ids())[(hdr.ref_ids()).size() - 1]] = (hdr.ref_ids()).size() - 1;
+                }
+            } break;
+
+            case 'R': /* RG (read group) tag */
             {
-                auto id_it = hdr.ref_dict.find(id);
+                std::pair<std::string, std::string> tmp{};
 
-                if (id_it == hdr.ref_dict.end())
-                    throw format_error{detail::to_string("Unknown reference name '", id, "' found in SAM header ",
-                                                         "(header.ref_ids(): ", hdr.ref_ids(), ").")};
+                parse_tag_value(get<0>(tmp));                           // read required ID tag
 
-                auto & given_ref_info = hdr.ref_id_info[id_it->second];
+                if (is_char<'\t'>(*it))                                 // read rest of the tags
+                {
+                    ++it;
+                    read_field(stream_view | views::take_until_or_throw(is_char<'\n'>), get<1>(tmp));
+                }
+                ++it;                                                   // skip newline
 
-                if (std::get<0>(given_ref_info) != std::get<0>(info))
-                    throw format_error{"Provided reference has unequal length as specified in the header."};
+                hdr.read_groups.emplace_back(std::move(tmp));
+            } break;
 
-                hdr.ref_id_info[id_it->second] = std::move(info);
-            }
-            else
+            case 'P': /* PG (program) tag */
             {
-                static_assert(!detail::is_type_specialisation_of_v<decltype(hdr.ref_ids()), std::deque>,
-                              "The range over reference ids must be of type std::deque such that "
-                              "pointers are not invalidated.");
+                typename alignment_file_header<ref_ids_type>::program_info_t tmp{};
 
-                hdr.ref_ids().push_back(id);
-                hdr.ref_id_info.push_back(info);
-                hdr.ref_dict[(hdr.ref_ids())[(hdr.ref_ids()).size() - 1]] = (hdr.ref_ids()).size() - 1;
-            }
-        }
-        else if (is_char<'R'>(*std::ranges::begin(stream_view)))         // RG (read group) tag
-        {
-            std::pair<std::string, std::string> tmp{};
+                parse_tag_value(tmp.id);                                // read required ID tag
 
-            parse_tag_value(get<0>(tmp));                                // read required ID tag
+                // The PN, CL, PP, DS, VN are optional tags and can be given in any order.
```

Comment:
As a side note, what happens if I pass `VR` or something as tag?
Because it seems that we only check for `N` but not the second character?

Reply:
There are quite a few places where the second letter is not checked when it is assumed to be unambiguous. I was also a little surprised by this, and I guess it is something we could change.
But it is unrelated to the other changes in this PR.

Reply:
I think I did not want to include to many `if` checks for the second character for performance reasons. We did not decide even yet whether we want to be very strict on format checking (e.g. samtools is not at all and swallows a lot).